CoreScriptsRoot/ServerCoreScripts/ServerChat/ChatChannel.lua

```diff
@@ -199,11 +207,33 @@ function methods:UnregisterProcessCommandsFunction(funcId)
 end
 
 function methods:SendSystemMessage(message)
+	self:AddMessageToHistoryLog(message, nil)
+
 	for i , speaker in pairs(self.Speakers) do
 		speaker:SendSystemMessage(message, self.Name)
 	end
 end
 
+function methods:RemoveExcessMessagesFromLog()
+	while (#self.ChatHistory > self.MaxHistory) do
+		table.remove(self.ChatHistory, 1)
+	end
+end
+
```

Comment:
Instead of calling `table.remove` each time, we could just call it once then call the variable. It can save a quite of CPU usage. So something like this,

> 
function methods:RemoveExcessMessagesFromLog()
local remove = table.remove
while (#self.ChatHistory > self.MaxHistory) do
remove(self.ChatHistory, 1) 
end
>